tonic/src/transport/server.rs

```diff
@@ -149,14 +164,26 @@ impl Server {
         self
     }
 
-    /// Consume this [`Server`] creating a future that will execute the server
-    /// on [`tokio`]'s default executor.
-    pub async fn serve<M, S>(self, addr: SocketAddr, svc: M) -> Result<(), super::Error>
+    /// Create a router with the `S` typed service as the first service.
+    ///
+    /// This will clone the `Server` builder and create a router that will
+    /// route around different services.
+    pub fn add_service<S>(&mut self, svc: S) -> Router<S, Unimplemented>
```

Comment:
Why does this need to take `&mut self` if it immediately clones `self`?

Reply:
Because you can create multiple routers from one `Server` config was the idea. It takes a &mut because it follows like the others.